src/app/app.ts

```diff
@@ -1,71 +1,125 @@
 import {Post} from 'mattermost-redux/types/posts';
-import {AppCall, AppContext} from 'mattermost-redux/types/apps';
+import {AppCall, AppContext, AppCallValues, AppCallResponse} from 'mattermost-redux/types/apps';
 
-import {Env, tryPromiseWithMessage, errorWithMessage} from '../utils';
+import {
+    newErrorCallResponseWithFieldErrors,
+    newOKCallResponse,
+    newOKCallResponseWithMarkdown,
+    newErrorCallResponseWithMessage,
+    FieldValidationErrors} from '../utils/call_responses';
+import {Env, tryPromiseWithMessage} from '../utils';
 
 import {newMMClient, newZDClient} from '../clients';
 
-import {configStore, oauthStore} from '../store';
+import {configStore} from '../store';
 
-import {newTicketFromForm, FieldValidationErrors} from './model';
+import {newTicketFromForm} from './ticketFromForm';
+import {newTriggerFromForm} from './triggerFromForm';
 
-class App {
-    createTicketFromPost = async (call: AppCall): Promise<FieldValidationErrors> => {
-        // get active mattermost user ID
-        const mmUserID = call.context.acting_user_id || '';
-        const zdToken = oauthStore.getToken(mmUserID);
-        if (!zdToken) {
-            throw new Error('Failed to get user access_token');
-        }
+export interface IApp {
+    createTicketFromPost(): Promise<AppCallResponse>;
+    createZDSubscription(): Promise<AppCallResponse>;
+}
+
+class App implements IApp {
+    context: AppContext
+    values: AppCallValues
 
+    constructor(call: AppCall) {
+        this.values = call.values;
+        this.context = call.context;
+    }
+    createTicketFromPost = async (): Promise<AppCallResponse> => {
         // get zendesk client for user
-        const zdClient = newZDClient(zdToken);
+        const zdClient = newZDClient(this.context);
 
         // create the ticket object from the form response
-        const [zdTicket, errors] = newTicketFromForm(call.values);
+        const [zdTicket, fieldErrors] = newTicketFromForm(this.values);
 
         // respond with errors
-        if (Object.keys(errors).length !== 0) {
-            return errors;
+        if (this.hasFieldErrors(fieldErrors)) {
+            return newErrorCallResponseWithFieldErrors(fieldErrors);
         }
 
         // create the ticket in Zendesk
-        const ticket = await tryPromiseWithMessage(zdClient.tickets.create(zdTicket), 'Failed to create Zendesk ticket');
+        const createReq = zdClient.tickets.create(zdTicket);
+        const ticket = await tryPromiseWithMessage(createReq, 'Failed to create Zendesk ticket');
```

Comment:
maybe rename this to `zdTicket` and rename the original `zdTicket` to smth. else? (to keep in sync with the `zdUser`, etc.)

Reply:
good nit catch. Done.